src/component/reducer/IgnoreActions.ts

```diff
@@ -34,11 +34,23 @@ const ignoreActions = [
   types.SHOW_J_GRAPH,
   types.SHOW_MULTIPLICTY_TREES,
   types.SHOW_RANGES_INTEGRALS,
+  types.CHANGE_FLOAT_MOLECULE_POSITION,
 ];
 
-function checkActionType(type: string): boolean {
+export function checkDataChangeActions(type: string): boolean {
   if (!ignoreActions.includes(type)) return true;
   return false;
 }
 
-export default checkActionType;
+const viewChangeActions = [
+  types.CHANGE_FLOAT_MOLECULE_POSITION,
+  types.FLOAT_MOLECULE_OVER_SPECTRUM,
+  types.SHOW_J_GRAPH,
+  types.SHOW_MULTIPLICTY_TREES,
+  types.SHOW_RANGES_INTEGRALS,
+];
+
```

Comment:
You should create a new useEffect that depends only on the viewState so that we don't need to specify the list of Actions.

The work you are currently doing (separation of view state from data state) is being done exactly to avoid this hack.

Reply:
thank you

Reply:
i used useEffect now instead of check Actions